### PR TITLE
[RW-9102][risk=low] Support preview Rstudio files using Terra Calhoun service

### PR DESCRIPTION
**Pull request by yonghaoy**
This is similar to our existing convert Jupyter, but call a different Terra service. 
Will have a followup PR/ticket to migrate Jupyter files

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

**Review by freemabd — APPROVED** [review]
This all seems reasonable to me. I'm assuming this is just following an already established pattern. LGTM

File: api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java [hunk 1]
```diff
@@ -328,7 +328,15 @@ public String getReadOnlyHtml(
...
     Blob blob = getBlobWithSizeConstraint(bucketName, notebookName);
-    return convertNotebookToHtml(blob.getContent());
+
+    if (NotebookUtils.isJupyterNotebook(notebookName)) {
+      return convertJupyterNotebookToHtml(blob.getContent());
+    } else if (NotebookUtils.isRstudioNotebook(notebookName)) {
+      return convertRstudioNotebookToHtml(blob.getContent());
+    } else {
+      throw new NotImplementedException(
+          String.format("%s type of file is not implemented yet", notebookName));
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java:338]: Let's add a little more context here: "Converting %s to read-only HTML is not supported" or similar

**Review by jmthibault79 — COMMENTED** [review]
Needs a rebase or merge

**Review by jmthibault79 — APPROVED** [review]

**Comment by yonghaoy** [comment]
https://github.com/all-of-us/workbench/pull/7700 will fix those breakages

**Pull request merged**